### PR TITLE
Put d:\home\site\tools folder on the path. Fixes #119.

### LINKED ISSUE #117: Should delete hostingstart.html
Today, this default file gets left over, which looks wrong when looking at raw files.

### LINKED ISSUE #119: Put d:\home\site\tools folder on the path
This will make it easier to drop shared tools for scenarios like ImageResize. Tools can also be dropped in each function folder, but this gives a shared option.

### PR DESCRIPTION
**Pull request by davidebbo**
Also, delete leftover hostingstart.html on start. Fixes #117.

### CODE REVIEW

File: src/WebJobs.Script.WebHost/App_Start/WebApiConfig.cs [hunk 1]
```diff
@@ -72,6 +83,33 @@ public static void Register(HttpConfiguration config, WebHostSettings settings =
             config.InitializeReceiveSalesforceWebHooks();
         }
 
+        private static void PrependFoldersToEnvironmentPath()
+        {
+            string home = Environment.GetEnvironmentVariable("HOME");
```
Comment (mathewc) [inline src/WebJobs.Script.WebHost/App_Start/WebApiConfig.cs]: Recommend adding an IsLocal to WebHostSettings

File: src/WebJobs.Script.WebHost/App_Start/WebApiConfig.cs [hunk 2]
```diff
@@ -72,6 +83,33 @@ public static void Register(HttpConfiguration config, WebHostSettings settings =
...
+                string toolsPath = Path.Combine(home, @"site\tools");
+                Directory.CreateDirectory(toolsPath);
+
+                var folders = new List<string>();
+                folders.Add(Path.Combine(home, @"site\tools"));
+
+                string path = Environment.GetEnvironmentVariable("PATH");
+                string additionalPaths = String.Join(";", folders);
+
+                // Make sure we haven't already added them. This can happen if the appdomain restart (since it's still same process)
```
Comment (mathewc) [inline src/WebJobs.Script.WebHost/App_Start/WebApiConfig.cs]: Could you introduce a static intitialize flag that you use to ensure you only do this init once and only once?

File: src/WebJobs.Script.WebHost/App_Start/WebApiConfig.cs [hunk 3]
```diff
@@ -80,7 +118,7 @@ private static WebHostSettings GetDefaultSettings()
             bool isLocal = string.IsNullOrEmpty(home);
             if (isLocal)
             {
-                settings.ScriptPath = Path.Combine(HostingEnvironment.ApplicationPhysicalPath, @"..\..\sample");
+                settings.ScriptPath = Path.Combine(HostingEnvironment.ApplicationPhysicalPath, @"D:\Dropbox\Functions\Samples");
```
Comment (mathewc) [inline src/WebJobs.Script.WebHost/App_Start/WebApiConfig.cs]: What is this Dropbox change?

File: src/WebJobs.Script.WebHost/App_Start/WebApiConfig.cs (hunk 1 above)
Comment (davidebbo) [inline src/WebJobs.Script.WebHost/App_Start/WebApiConfig.cs]: We could, but note that the logic actually uses HOME beyond detection. So we'd really need to add HomeFolder to `WebHostSettings`. And note sure is that would push changes in a bunch of other places.

File: src/WebJobs.Script.WebHost/App_Start/WebApiConfig.cs (hunk 2 above)
Comment (davidebbo) [inline src/WebJobs.Script.WebHost/App_Start/WebApiConfig.cs]: No, statics don't survive AppDomain restarts.

File: src/WebJobs.Script.WebHost/App_Start/WebApiConfig.cs (hunk 3 above)
Comment (davidebbo) [inline src/WebJobs.Script.WebHost/App_Start/WebApiConfig.cs]: Crap, good catch! That's my local testing thing :) Reversed.

File: src/WebJobs.Script.WebHost/App_Start/WebApiConfig.cs (hunk 1 above)
Comment (mathewc) [inline src/WebJobs.Script.WebHost/App_Start/WebApiConfig.cs]: Right, you need the value too. Forget it I guess.

**Comment by mathewc** [comment]
:ship:

**Comment by mathewc** [comment]
:shipit:

**Comment by davidebbo** [comment]
Merged.

**Pull request closed without merging**